runtime/jcl/common/reflecthelp.c

```diff
@@ -105,6 +117,9 @@ getAnnotationDataFromROMMethodHelper(struct J9VMThread *vmThread, J9Method *ramM
 	U_32 *annotationData = getAnnotationDataFromROMMethod(romMethod);
 	if (NULL != annotationData) {
 		result = getAnnotationDataAsByteArray(vmThread, annotationData);
+		if (NULL != result) {
```

Comment:
Instead of doing the NULL check and append in the callers, you can change `getAnnotationDataAsByteArray()` to take constant pool as an new input parameter. If it is not NULL, it will be appended to the array inside `getAnnotationDataAsByteArray()`.